src/Uno.UI/UI/Xaml/FontManifestHelpers.cs

```diff
@@ -0,0 +1,62 @@
+﻿#nullable enable
+
+using System;
+using System.IO;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Windows.UI.Text;
+
+namespace Uno.UI.Xaml.Media;
+
+internal static class FontManifestHelpers
+{
+	private static readonly JsonSerializerOptions _options = new JsonSerializerOptions()
+	{
+		AllowTrailingCommas = true,
+		PropertyNamingPolicy = JsonNamingPolicy.SnakeCaseLower,
+		ReadCommentHandling = JsonCommentHandling.Skip,
+		Converters =
+		{
+			new JsonStringEnumConverter(),
+		},
+		TypeInfoResolver = FontManifestSerializerContext.Default,
+	};
+
+	internal static string GetFamilyNameFromManifest(Stream jsonStream, FontWeight weight, FontStyle style, FontStretch stretch)
+	{
+		var manifest = JsonSerializer.Deserialize<FontManifest>(jsonStream, _options);
+		return GetFamilyNameFromManifest(manifest, weight, style, stretch);
+	}
+
+	internal static string GetFamilyNameFromManifest(FontManifest? manifest, FontWeight weight, FontStyle style, FontStretch stretch)
+	{
+		if (manifest?.Fonts is null || manifest.Fonts.Length == 0)
+		{
+			throw new ArgumentException("Font manifest file is incorrect.");
+		}
+
+		var bestSoFar = manifest.Fonts[0];
```

Comment:
We should consider implement this as per https://www.w3.org/TR/css-fonts-3/#font-style-matching. See 5.2 point 4